docs/build.gradle

```diff
@@ -16,10 +16,29 @@ apply plugin: 'org.antora'
 
 description = 'Micrometer Docs'
 
+// Spring 6 requires Java 17+
+// skip this module when building with jdk <17
+if (!javaLanguageVersion.canCompileOrRun(17)) {
+    project.tasks.configureEach { task -> task.enabled = false }
```

Comment:
I assume this is okay for the docs module since we run it with JDK 17 (according to [this](https://github.com/micrometer-metrics/micrometer/actions/runs/11266458878/workflow#L25-L29)), but let me know if I'm missing any reason this would be a problem.